pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -148,6 +164,36 @@ func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
 	}
 }
 
+func loadFileOrDir(fileOrDir string) ([]string, error) {
+	stat, err := os.Stat(fileOrDir)
+	if err != nil {
+		return []string{}, err
+	}
+	if stat.IsDir() {
+		return loadConfigDir(fileOrDir)
+	}
+	return []string{fileOrDir}, nil
```

Comment:
> Is there a function / reference where your sort order is defined? I would have sorted 12_foo before 1_foo.

Oh, and there was no sorting before.  The user explicitly listed files in an order.  This is adding the need for sorting.